apps/prairielearn/src/ee/pages/administratorInstitutionSaml/administratorInstitutionSaml.html.ts

```diff
@@ -297,45 +298,89 @@ ${samlProvider?.certificate ?? ''}</textarea
+                  <button
+                    class="btn btn-danger"
+                    type="button"
+                    data-toggle="modal"
+                    data-target="#deleteModal"
+                  >
+                    Delete SAML configuration
+                  </button>
                 </p>
-              </div>
-              <div class="modal-footer">
-                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+
+                <h2 class="h4">Decode SAML assertion</h2>
+
                 <form method="POST">
+                  <div class="form-group">
+                    <label for="encodedAssertion">Encoded assertion</label>
+                    <textarea
+                      class="form-control"
+                      id="encodedAssertion"
+                      rows="10"
+                      name="encoded_assertion"
+                      aria-describedby="encodedAssertionHelp"
+                    ></textarea>
+                    <small class="form-text text-muted">
+                      This should be raw base64-encoded data from the
+                      <code>SAMLResponse</code> parameter in the POST request from the IdP.
+                    </small>
+                  </div>
+
+                  <div class="form-group form-check">
+                    <input
+                      type="checkbox"
+                      class="form-check-input"
+                      id="strictMode"
+                      name="strict_mode"
+                      value="1"
+                      aria-describedBy="strictModeHelp"
+                    />
+                    <label class="form-check-label" for="strictMode">Strict mode</label>
+                    <small id="strictModeHelp" class="form-text text-muted mt-0">
+                      Forces "validate audience", "require signed assertions", and "require signed
+                      response" to be enabled.
+                    </small>
+                  </div>
+
                   <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
-                  <button class="btn btn-danger" type="submit" name="__action" value="delete">
-                    Delete SAML configuration
+                  <button
+                    class="btn btn-primary"
+                    id="decodeAssertionButton"
+                    type="button"
+                    name="__action"
+                    value="decodeAssertion"
+                    hx-post="${resLocals.urlPrefix}/saml"
+                    hx-target="#decodedAssertion"
+                    hx-swap="innerHTML show:top"
+                  >
+                    Decode
                   </button>
+                  <div id="decodedAssertion"></div>
                 </form>
-              </div>
-            </div>
-          </div>
-        </div>
+              `
+            : ''}
+        </main>
+
+        ${Modal({
```

Comment:
Thanks for pulling this out of the main HTML block. Could even go one step further and make it a separate function that gets included early in the page.

Reply:
Done in 3ade77d.